[HttpBinding] Http commands fail silently for illegal URLs

## Current Behavior

The HTTP binding fails to send an HTTP command when it encounters a bad/malformed/illegal URL, but does not report any error in the log files to indicate that the failure occurred.

(This problem does not occur in OpenHAB 2.x)
## Expected Behavior

The binding should log the error so that the user can tell the command failed.
## Possible Solution

The problem occurs in the `org.openhab.io.net.http.HttpUtil` class.  My proposal is to catch and log the error.  I have the code change almost done...
## Steps to Reproduce (for bugs)
1. Create an HTTP "out" binding with an illegal URL
   eg.

`{http=">[*:POST:http://www.google.com/home/lights/23873/?status=%2$s&type=\"text\"]"}`
## 


<bountysource-plugin>
---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/37992514-httpbinding-http-commands-fail-silently-for-illegal-urls?utm_campaign=plugin&utm_content=tracker%2F482064&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F482064&utm_medium=issues&utm_source=github).
</bountysource-plugin>

The example URL I gave is illegal because the double quote character `"` is not allowed:
>http://www.google.com/home/lights/23873/?status=%2$s&type=\"text\"

So when the [Apache code](https://github.com/openhab/openhab/blob/1.8/bundles/io/org.openhab.io.net/src/main/java/org/openhab/io/net/http/HttpUtil.java#L36) receives the URL below, it throws an exception, which never gets logged.

>http://www.google.com/home/lights/23873/?status=%2$s&type="text"

And here you can see the thrown exception resulting from adding a try/catch block:
>21:44:13.094 [ERROR] [[g.openhab.io.net.http.HttpUtil:351 ](https://github.com/openhab/openhab/blob/1.8/bundles/io/org.openhab.io.net/src/main/java/org/openhab/io/net/http/HttpUtil.java#L342) ] - Failed to create POST method: Invalid uri 'http://www.google.com/home/lights/23873/?status=ON&type="text"': Invalid query

The only legal way to have double quotes in a URL is to escape them, which means to percent-encode them.  However, I'm not aware of any reason to include double quotes in a query like this.

--- a/bundles/io/org.openhab.io.net/src/main/java/org/openhab/io/net/http/HttpUtil.java
+++ b/bundles/io/org.openhab.io.net/src/main/java/org/openhab/io/net/http/HttpUtil.java
@@ -41,7 +41,6 @@ import org.apache.commons.lang.StringUtils;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-
 /**
  * Some common methods to be used in both HTTP-In-Binding and HTTP-Out-Binding
  * 

